wpiutil/src/test/java/edu/wpi/first/util/struct/ProceduralStructifierTest.java

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.util.struct;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import edu.wpi.first.util.struct.ProceduralStructifier.ProcEnumStruct;
+import edu.wpi.first.util.struct.ProceduralStructifier.ProcRecordStruct;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import org.junit.jupiter.api.Test;
+
+class ProceduralStructifierTest {
+  public record CustomRecord(int int32, boolean bool, double float64, char character, short int16)
+      implements StructSerializable {
+    public static CustomRecord create() {
+      return new CustomRecord(42, true, Math.PI, 'a', (short) 16);
+    }
+
+    public static final Struct<CustomRecord> struct = ProcRecordStruct.generate(CustomRecord.class);
+
+    @Override
+    public final boolean equals(Object arg) {
+      if (arg == this) {
+        return true;
+      }
+      if (!(arg instanceof CustomRecord)) {
+        return false;
+      }
+      CustomRecord other = (CustomRecord) arg;
+      return int32 == other.int32
```

Comment:
Use pattern matching.
```suggestion
      return arg instanceof CustomRecord other
          && int32 == other.int32
```